src/app/components/timer/timer.component.ts

```diff
@@ -35,6 +37,13 @@ export class TimerComponent {
       this.seconds = (this.game.turnEnds - Date.now()) / 1000;
       this.clock = this.fancyTimeFormat(this.seconds);
 
+      if (this.seconds <= 0 && this.room.enforceTimer) {
+        this.gameService.updateGame(this.game.id, {
+          status: this.game.status === GameStatus.BLUES_TURN ? GameStatus.REDS_TURN : GameStatus.BLUES_TURN,
+          turnEnds: Date.now() + (this.room.timer * 1000)
+        });
+      }
```

Comment:
Hey @TangoCardBob, I was wondering where the best place for this logic to live would be. I'm worried about asynchronous behavior happening from several browsers. The timer is rendered for everyone, so if we have 10 people in a game, this logic may happen up to 10 times. Do you think it could ever result in the game status being wrong?

What if we only had one player (maybe the current team's spymaster) run this logic to end their own team's turn? Or do you think this is a non-issue?